great_expectations_cloud/agent/models.py

```diff
@@ -38,6 +38,7 @@ class RunMissingnessDataAssistantEvent(RunDataAssistantEvent):
 
 class RunCheckpointEvent(EventBase):
     type: Literal["run_checkpoint_request"] = "run_checkpoint_request"
+    datasource_name: Union[str, None] = None
```

Comment:
Nullability of param discussed here: https://github.com/greatexpectationslabs/mercury/pull/2772#discussion_r1481944407